pyqtgraph/graphicsItems/PColorMeshItem.py

```diff
@@ -0,0 +1,253 @@
+            otherwise x and y will be used as polygons vertices coordinates as:
+
+            (x[i+1, j], y[i+1, j])           (x[i+1, j+1], y[i+1, j+1])
+                                +---------+
+                                | z[i, j] |
+                                +---------+
+                (x[i, j], y[i, j])           (x[i, j+1], y[i, j+1])
+            "ASCII from: https://matplotlib.org/3.2.1/api/_as_gen/
+                         matplotlib.pyplot.pcolormesh.html".
+
+        """
+
+        # Prepare data
+        cd = self._prepareData(args)
+
+        # Has the view bounds changed
+        shapeChanged = False
+        if self.qpicture is None:
+            shapeChanged = True
+        elif len(args)==1:
+            if args[0].shape[0] != self.x[:,1][-1] or args[0].shape[1] != self.y[0][-1]:
+                shapeChanged = True
+        elif len(args)==3:
+            if np.any(self.x != args[0]) or np.any(self.y != args[1]):
+                shapeChanged = True
+
+        profile = debug.Profiler()
+
+        self.qpicture = QtGui.QPicture()
+        p = QtGui.QPainter(self.qpicture)
+        
+        # We set the pen of all polygons once
+        if self.edgecolors is None:
+            p.setPen(QtGui.QColor(0, 0, 0, 0))
+        else:
+            p.setPen(fn.mkPen(self.edgecolors))
+
+        ## Prepare colormap
+        # First we get the LookupTable
+        pos   = [i[0] for i in Gradients[self.cmap]['ticks']]
+        color = [i[1] for i in Gradients[self.cmap]['ticks']]
+        cmap  = ColorMap(pos, color)
+        lut   = cmap.getLookupTable(0.0, 1.0, 256)
+        # Second we associate each z value, that we normalize, to the lut
+        norm  = self.z - self.z.min()
+        norm = norm/norm.max()
+        norm  = (norm*(len(lut)-1)).astype(int)
+        
+        # Go through all the data and draw the polygons accordingly
+        for xi in range(self.z.shape[0]):
+            for yi in range(self.z.shape[1]):
+                
+                # Set the color of the polygon first
+                # print(xi, yi, norm[xi][yi])
+                c = lut[norm[xi][yi]]
+                p.setBrush(QtGui.QColor(c[0], c[1], c[2]))
+
+                polygon = QtGui.QPolygonF(
+                    [QtCore.QPointF(self.x[xi][yi],     self.y[xi][yi]),
+                     QtCore.QPointF(self.x[xi+1][yi],   self.y[xi+1][yi]),
+                     QtCore.QPointF(self.x[xi+1][yi+1], self.y[xi+1][yi+1]),
+                     QtCore.QPointF(self.x[xi][yi+1],   self.y[xi][yi+1])]
+                )
+
+                # DrawConvexPlygon is faster
+                p.drawConvexPolygon(polygon)
+
+
+        p.end()
+        self.update()
+
+        if shapeChanged:
+            self.informViewBoundsChanged()
+
+
+
+    def paint(self, p, *args):
+        profile = debug.Profiler()
+        if self.z is None:
+            return
+
+        profile('p.drawPicture')
+        p.drawPicture(0, 0, self.qpicture)
```

Comment:
you may be able to redo this as something like 

```python
if self.qpicture is not None:
    self.qpicture.play(p)
```

The implementation taken from TargetItem.py https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/TargetItem.py#L73-L76

Reply:
wait, that probably does nothing, was reading the docs:

```
This function does exactly the same as QPainter::drawPicture() with (x, y) = (0, 0).
```

Reply:
Ok, so this is clearly it because if I add a coefficient in the `boundingRect` method to return, for instance,  `QtCore.QRectF(0., 0., float(self.width())*2., float(self.height())*2.)` then the artifacts disappear.

I tried fix to make it better but so far it didn't work well...